### PR TITLE
Libra API v1.2

### PR DESCRIPTION
**Pull request by Tomas-Per**
* API changes
  * The get method for computers is now split in two: laptops and desktops have their own GET actions with different API routes. Furthermore, each of these GET actions are further split into chunks (pages)
     * Each chunk (page) is 20 items
     * Chunks (pages) are accessed with numbers from 1 to n, where 0 is left for getting all of the data (see swagger doc for better clarity)
     * This will enhance performance of fetch operations, since less data will have to be accessed at one given time
   * Added a forgot password controller action
     * This HTTP PUT action sets a given (determined by the given email) user's recovery password to a random value generated with `System.Security.Cryptography.RNGCryptoServiceProvider`
   * The UpdateUser method now sets the recovery password to null 
   * Modified/Fixed routing
   * Added XML comments to controllers, which are shown in swagger UI
* Database changes
  * User model
    * Added RecoveryPassword
    * Added RecoveryDate

### CODE REVIEW

File: DataContent/DAL/Repositories/UserRepository.cs [hunk 1]
```diff
@@ -96,9 +97,19 @@ public async Task<User> UpdateUserAsync(User user)
...
                 updatedUser.Password = new ScryptEncoder().Encode(user.Password);
+                updatedUser.RecoveryPassword = null;
             }
             await _context.SaveChangesAsync();
             return updatedUser;
         }
+
+        public async Task<User> ForgotPasswordAsync(string email)
+        {
+            var user = await _context.Users.Where(x => x.Email == email).FirstOrDefaultAsync();
```
Comment (Ugniusst) [inline DataContent/DAL/Repositories/UserRepository.cs:108]: I think there we need to check, if there was at least 30s or more between recoveries before adding new recovery password

Comment (Tomas-Per) [reply DataContent/DAL/Repositories/UserRepository.cs:108]: Done

File: DataContent/DAL/Repositories/UserRepository.cs [hunk 2]
```diff
@@ -96,9 +97,23 @@ public async Task<User> UpdateUserAsync(User user)
...
+            if(user.RecoveryDate.AddSeconds(30) < DateTime.Now)
+            {
+                user.RecoveryPassword = PasswordHelper.RandomString(16);
+                user.RecoveryDate = DateTime.Now;
+                await _context.SaveChangesAsync();
+            }
+            
+            return user;
```
Comment (Ugniusst) [inline DataContent/DAL/Repositories/UserRepository.cs:117]: should throw exception if recoverydate was less than 30s, so we could show an error to user later

Comment (Tomas-Per) [reply DataContent/DAL/Repositories/UserRepository.cs:117]: Done

**Review by Ugniusst — APPROVED** [review]

**Review by Filari02 — APPROVED** [review]

**Pull request merged**